### PR TITLE
User management

### PR DESCRIPTION
**Pull request by alexkaserbacher**
- Adds Atlassian Crowd as authentication provider to ApiCenter
- Adds support for JWT to authenticate user requests
- Adds frontend functionality to login and logout users

### CODE REVIEW

File: backend/src/main/kotlin/com/tngtech/apicenter/backend/config/SecurityCrowdConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    WebSecurityConfigurerAdapter() {
+
+    @Value("\${jwt.secret}")
+    private lateinit var jwtSecuritySecret: String;
+
+    override fun configure(httpSecurity: HttpSecurity) {
+        httpSecurity
+            .cors()
+            .and()
+            .csrf().disable()
```
Comment (fdw) [inline backend/src/main/kotlin/com/tngtech/apicenter/backend/config/SecurityCrowdConfiguration.kt]: Why?

File: backend/src/main/kotlin/com/tngtech/apicenter/backend/config/SecurityCrowdConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+@EnableWebSecurity
+@Configuration
+class SecurityCrowdConfiguration @Autowired constructor(private val userHandler: UserHandler) :
+    WebSecurityConfigurerAdapter() {
+
+    @Value("\${jwt.secret}")
+    private lateinit var jwtSecuritySecret: String;
+
+    override fun configure(httpSecurity: HttpSecurity) {
```
Comment (fdw) [inline backend/src/main/kotlin/com/tngtech/apicenter/backend/config/SecurityCrowdConfiguration.kt]: This method configures the ApiCenter itself, while the ones below only concern the configuration in regard to Crowd, right? Maybe split this file into two?

File: backend/src/main/kotlin/com/tngtech/apicenter/backend/domain/handler/SessionHandler.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            searchUser
+        }
+
+        val userToStore = if (storedUser.email != searchUser.email || storedUser.username != searchUser.username) {
+            User(storedUser.id, searchUser.email, searchUser.username, "crowd", searchUser.externalId)
+        } else {
+            storedUser
+        }
+
+        if (userHandler.checkExistenceByOrigin(searchUser)) userHandler.store(userToStore)
```
Comment (fdw) [inline backend/src/main/kotlin/com/tngtech/apicenter/backend/domain/handler/SessionHandler.kt]: Please use braces

File: backend/src/main/resources/api-definition.json [hunk 4]
```diff
@@ -27,6 +27,16 @@
...
+            "in": "header",
+            "name": "Authorization",
+            "description": "JWT to authenticate the user, preceded with the string 'Bearer '",
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
         "responses": {
           "200": {
```
Comment (fdw) [inline backend/src/main/resources/api-definition.json]: Now it should also be possible to get 403s, right?

File: backend/src/main/kotlin/com/tngtech/apicenter/backend/config/SecurityCrowdConfiguration.kt (hunk 1 above)
Comment (alexkaserbacher) [reply backend/src/main/kotlin/com/tngtech/apicenter/backend/config/SecurityCrowdConfiguration.kt]: Don't we want to allow requests from multiple sources (without the need to issue a CSRF token)? Client, CI-Server, Gradle Plugins etc.

File: backend/src/main/resources/api-definition.json (hunk 4 above)
Comment (alexkaserbacher) [reply backend/src/main/resources/api-definition.json]: Yes, it's already in there ;)

**Pull request merged**